tests/Service/PromotionsService/AbstractPromotionServiceTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Tests\BBC\ProgrammesPagesService\Service\PromotionsService;
+
+use BBC\ProgrammesPagesService\Domain\Entity\Brand;
+use BBC\ProgrammesPagesService\Domain\Entity\Promotion;
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use BBC\ProgrammesPagesService\Service\PromotionsService;
+use Tests\BBC\ProgrammesPagesService\AbstractServiceTest;
+
+abstract class AbstractPromotionServiceTest extends AbstractServiceTest
+{
+    public function setUp()
+    {
+        $this->setUpCache();
+        $this->setUpRepo('PromotionRepository');
+        // override getDomainModel() in mapper with getDomainModelFromDbData()
+        $this->setUpMapper('PromotionMapper', 'getDomainModelFromDbData');
+    }
+
+    /**
+     * @param mixed[] $dbPromotion
+     */
+    protected function getDomainModelFromDbData(array $dbPromotion): Promotion
```

Comment:
It's a tool for mocking out the Mapper, otherwise the service tests would end up testing the Service and Mapper functionality.

This is the style that we use for all Service tests so if we change it, we should do it all at once rather than in this PR.